### DISCUSSION TITLE
issue: Watch method returns values out of sync from useFieldArray. Watch only triggered when appended to or removed

### DISCUSSION THREAD

**Discussion by mckenziecallum** [react-hook-form/react-hook-form#9994] [Q&A]
### Version Number

7.43.2

### Codesandbox/Expo snack

https://codesandbox.io/s/issue-usefieldarray-watch-doesnt-trigger-osxtzx

### Steps to reproduce

1. Go to the codesandbox
2. Open console
3. Click one of the text fields
4. Modify the value
5. Notice no log line is printed for watch
6. Click append
7. Notice a log line is printed from watch with new value for modified field

### Expected behaviour

When one of the fields in the `useFieldArray` is modified, watch should be triggered with the new value. We should be able to subscribe to values of a nested field to allow for use externally. This used to happen in older versions of ReactHookForm.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment]
@PixelYeti 

The `watch('field_name')` method with parameter as field name is expected to be return value for each render of the component where it's triggered. Only `watch(callbackFunction)` with parameter as a callback function will subscribe on change of fields.

Could you share what version > v7.0.0 used to support watch on change on field name string ? As I remember, the behavior is not changed from the beginning of v7.

**Reply by mckenziecallum** [reply]
@leapful

Sorry it wasn't on a version > v7. I found an old issue with a similar problem but that was v5.

So is the solution to use watch(callbackFunction) for what I want to achieve?

**Reply by leapful** [reply]
@PixelYeti 

You can use `useWatch` hook instead of `watch(callbackFunction)`.
